thriftbp/server.go

```diff
@@ -0,0 +1,41 @@
+package thriftbp
+
+import (
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"github.com/reddit/baseplate.go/log"
+)
+
+// ServerConfig is the arg struct for NewThriftServer.
+type ServerConfig struct {
+	// The endpoint address of your thrift service
+	Addr string
+
+	// The timeout for the underlying thrift.TServerSocket transport.
+	Timeout time.Duration
+
+	// A log wrapper that is passed to the "Wrap" call and server.SetLogger
+	Logger log.Wrapper
+}
+
+// NewThriftServer returns a thrift.TSimpleServer using the THeader transport
+// and protocol to serve the given BaseplateProcessor which is wrapped with the
+// given Middlewares.
+func NewThriftServer(cfg ServerConfig, processor BaseplateProcessor, middlewares ...Middleware) (*thrift.TSimpleServer, error) {
+	transport, err := thrift.NewTServerSocketTimeout(cfg.Addr, cfg.Timeout)
+	if err != nil {
+		return nil, err
+	}
+
+	server := thrift.NewTSimpleServer4(
+		Wrap(processor, cfg.Logger, middlewares...),
```

Comment:
I didn't notice this in the other PR, but logger is unused in `Wrap` implementation so it should be removed from `Wrap`'s function signature: https://github.com/reddit/baseplate.go/blob/e21d93469801525d38a8848ee56d74ef83db3af5/thriftbp/middlewares.go#L71-L80